### PR TITLE
4421react_dev_warning

### PR DESCRIPTION
**Pull request by felixgondwe**
-added plugins to port react production build
-updated react version to v15.4.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,6 +83,11 @@
     "htmltidy": "0.0.6",
     "loose-envify": "^1.2.0",
     "mocha": "^2.3.4",
+    "react": "^15.4.0",
```
Comment (ragingsquirrel3) [inline package.json]: I think some deps are repeated.  Please consolidate.

File: package.json [hunk 2]
```diff
@@ -77,6 +83,11 @@
...
+    "react-dom": "^15.4.0",
+    "uglify-js": "^3.0.15",
+    "uglifyify": "^3.0.4",
+    "webpack": "^2.6.1",
```
Comment (ragingsquirrel3) [inline package.json]: Why webpack?  Please make sure unneeded deps are not here.

File: Gruntfile.js [hunk 3]
```diff
@@ -142,7 +145,11 @@ module.exports = function(grunt) {
                     browserifyOptions: {
                         debug: false
                     },
-                    transform: ["babelify", ["loose-envify", { "NODE_ENV": "production" }]]
+                    transform: [
+                        "babelify",
+                        ["uglifyify",{global:true}],
```
Comment (ragingsquirrel3) [inline Gruntfile.js]: There is already an uglify task being run, so if you want to use this version, please remove the other task.

File: package.json (hunk 1 above)
Comment (felixgondwe) [inline package.json]: Cleaned up and consolidated the deps.

File: package.json (hunk 2 above)
Comment (felixgondwe) [inline package.json]: its a babel dependency, I have however reverted to webpack 1.* as it was before.

Comment (ragingsquirrel3) [inline package.json]: I don't believe webpack is currently listed as an explicit dependency in package.json.  I don't believe it's necessary to include here, but let me know if you're having a problem without it.

File: Gruntfile.js (hunk 3 above)
Comment (felixgondwe) [inline Gruntfile.js]: Resolved to keep uglifyify and uglify to get smaller output for performance purposes.

**Pull request merged**